### ISSUE TITLE
go/doc: example should not include test functions

### ISSUE THREAD

**Issue by rhcarvalho** [golang/go#37900]
### What version of Go are you using (`go version`)?

<pre>
$ go version
go1.14
</pre>

### Does this issue reproduce with the latest release?

Yes and tip as well.

### What operating system and processor architecture are you using (`go env`)?

Not revelant.

### What did you do?

Visited https://golang.org/pkg/errors/#example_As.

Then visited https://godoc.org/errors#example-As and https://pkg.go.dev/errors/?tab=doc#example-As to compare, same problem.

### What did you expect to see?

As per documented in https://golang.org/pkg/testing/#hdr-Examples (emphasis mine):

> The entire test file is presented as the example when it contains a single example function, at least one other function, type, variable, or constant declaration, **and no test or benchmark functions**.

I expected to see the example for using `errors.As`, a rather small snippet containing:

```go
   if _, err := os.Open("non-existing"); err != nil {
        var pathError *os.PathError
        if errors.As(err, &pathError) {
            fmt.Println("Failed at path:", pathError.Path)
        } else {
            fmt.Println(err)
        }
    }
```

As defined in https://golang.org/src/errors/wrap_test.go?s=4623:4893#L241.

### What did you see instead?

A large amount of code containing tests defined in https://golang.org/src/errors/wrap_test.go, with the actual example code transformed into a `func main` at the very bottom.

![image](https://user-images.githubusercontent.com/88819/76854675-d4326900-684f-11ea-98e0-b925fd7df23e.png)

![image](https://user-images.githubusercontent.com/88819/76856411-815ab080-6853-11ea-9586-7232d9781bb2.png)

**Comment by agnivade** [comment]
This is fixed now on tip: https://tip.golang.org/pkg/errors/#example_As

**Comment by rhcarvalho** [comment]
@agnivade thanks for the update. By any chance, do you know what commit fixed it? Just curious to know what the problem was.

**Comment by agnivade** [comment]
https://github.com/golang/go/commit/c4961dc247ca39c251a5a3c80ebfe59609b4e669#diff-23c81667acc0871b0276b715d353197f

**Comment by rhcarvalho** [comment]
FTR, this was reported at a later date and fixed in #38409. Happy to see it was backported to go1.14.3!
